ssh/tailssh/tailssh.go

```diff
@@ -124,6 +148,47 @@ func (srv *server) mayForwardLocalPortTo(ctx ssh.Context, destinationHost string
 	return ss.action.AllowLocalPortForwarding
 }
 
+// askForPubKey reports whether the SSH server, during the auth negotiation
+// phase, should requires that the client send an SSH public key. (or, more
+// specifically, that "none" auth isn't acceptable)
+func (srv *server) askForPubKey(sshUser string, localAddr, remoteAddr net.Addr) bool {
+	pol, ok := srv.sshPolicy()
+	if !ok {
+		return false
+	}
+	a, ci, _, err := srv.evaluatePolicy(sshUser, localAddr, remoteAddr, nil)
+	if err == nil && (a.Accept || a.HoldAndDelegate != "") {
+		// Policy doesn't require a public key.
+		return false
+	}
+
+	// Is there any rule that looks like it'd require a public key for this
+	// sshUser?
+	now := time.Now()
+	for _, r := range pol.Rules {
+		if r.RuleExpires != nil && now.After(*r.RuleExpires) {
```

Comment:
add `IsExpired()` to the Rule?

Reply:
we generally don't have code in the tailcfg package.

also, having that would mean it would need to use time.Now() and we don't here.

but I'll see if there's anywhere to pull it out

Reply:
done